frontend/pkg/frontend/frontend.go

```diff
@@ -258,12 +261,36 @@ func (f *Frontend) ArmResourceRead(writer http.ResponseWriter, request *http.Req
 
 	// URL path is already lowercased by middleware.
 	resourceID := request.URL.Path
-	cluster, found := f.cache.GetCluster(resourceID)
-	if !found {
-		writer.WriteHeader(http.StatusNotFound)
+	subscriptionID := request.PathValue(PathSegmentSubscriptionID)
+	doc, err := f.dbClient.GetClusterDoc(ctx, resourceID, subscriptionID)
+	if err != nil {
+		if errors.Is(err, database.ErrNotFound) {
+			f.logger.Error(fmt.Sprintf("existing document not found for cluster: %s", resourceID))
+			writer.WriteHeader(http.StatusNoContent)
+			return
+		} else {
+			f.logger.Error(err.Error())
+			writer.WriteHeader(http.StatusInternalServerError)
+			return
+		}
+	}
+
+	cluster, err := f.conn.ClustersMgmt().V1().Clusters().Cluster(doc.ClusterID).Get().Send()
```

Comment:
Maybe not for this PR and maybe not useful for the team, but I think it could be a good idea to put an interface (like ClustersController or something similar) between frontend and CS to provide a contract where the type the interface understands is _api.HCPOpenShiftCluster_.

That way, we could run the frontend without CS if we want with a simple in-memory struct that implements the interface. By doing so, we can run the frontend and try it without explicitly depending on CS details. To do that, we would move all the CS specifics that are in frontend.go to its package, and in NewFrontend() pass an interface as a parameter that gets assigned to a field of the frontend. 

Then we could use an env variable (like LOCAL_CS="true" with a default value of "false") that would instantiate a fake CS instead of the "real one", and we would control that in frontend/cmd/cmd.go where we are calling NewFrontend(). That way we can decouple frontend and CS and just focus on the contract in the frontend code.

Reply:
> That way, we could run the frontend without CS if we want with a simple in-memory struct that implements the interface.

I was saying this exact thing to Michael yesterday about this, not requiring CS would be nice. Follow up PR/Card seems like a good place, i can get one created for backlog and see what folks think